bazel/sh_test_wrapper.sh

```diff
@@ -6,4 +6,4 @@
 
 cd $(dirname "$0")
 
-"$@"
+"./$@"
```

Comment:
If I understand correctly, this won't be necessary after bazel 3.1.0. If I've got that right, we should a todo here (in our standard style with a github username) to revert this once that's in place.

Reply:
@zuercher actually, this will be necessary after bazel 3.1.0, currently with bazel 3.0.0 and below the `sh_test_wrapper.sh` does not have the correct execution environment/runfiles setup/symlinks so the test file the `envoy_sh_test` rule sets up cannot find the test script to invoke at all (even with `./`), with bazel 3.1.0 and above, the runfiles are set up correctly for the sh_test rule, but running the `sh_test_wrapper.sh` script as-is causes the `command not found` error, adding `./` fixes that for bazel 3.1.0

I can re-clarify in the issue description if needed to make that clearer

Reply:
this is a minimal example that demonstrates the issue (linked in the bazel issue above): https://github.com/greenhouse-org/bazel-issue-repro/blob/c7ebe740e6cf12b58c7c7e156d9f5d2670329aad/sh_test_symlinks/README.md